src/main/java/com/bookshop/mapper/BookMapper.java

```diff
@@ -1,17 +1,27 @@
 package com.bookshop.mapper;
 
 import com.bookshop.config.MapperConfig;
-import com.bookshop.dto.BookDto;
-import com.bookshop.dto.CreateBookRequestDto;
+import com.bookshop.dto.book.BookDto;
+import com.bookshop.dto.book.BookDtoWithoutCategoryIds;
+import com.bookshop.dto.book.CreateBookRequestDto;
 import com.bookshop.model.Book;
+import com.bookshop.model.Category;
 import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
 import org.mapstruct.MappingTarget;
 
 @Mapper(config = MapperConfig.class)
 public interface BookMapper {
+    @Mapping(source = "categories", target = "categoriesIds")
     BookDto toDto(Book book);
 
     Book toBook(CreateBookRequestDto requestDto);
 
     void updateBook(CreateBookRequestDto dto, @MappingTarget Book book);
+
+    BookDtoWithoutCategoryIds toDtoWithoutCategories(Book book);
+
+    default Long categoryToId(Category category) {

```

Comment:
I still can't catch the logic of this method, I think it has the right to life. But think @AfterMapping over it, and if it's needed there